Sources/Sentry/SentryCrashIntegration.m

```diff
@@ -190,6 +191,21 @@ - (void)configureScope
                     userInfo[@"release"] = self.options.releaseName;
                     userInfo[@"dist"] = self.options.dist;
 
+                    NSMutableArray *attachments = [NSMutableArray new];
+                    for (SentryAttachment *attachment in [scope attachments]) {
+                        NSMutableDictionary *attachmentDict = [NSMutableDictionary new];
+                        attachmentDict[@"data"] =
+                            [[NSString alloc] initWithData:attachment.data
```

Comment:
Is this getting the attachment NSData and creating a UTF8 String out of it?

Reply:
Yep exactly, so SentryCrash can store it to disk. The downside of this approach is that we have attachments with data twice in memory.

Reply:
This isn't going to work, you're getting a byte array and encoding it.
You'd need to base64 or use another way to serialize that byte array instead.

Reply:
I honestly think this isn't good way to approach this.

I recommend simply passing a file `path` and reading the file after restart. Clearly documenting the behavior. Even if that means we take the byte[], create a temp file and use it later. But getting a whole byte blob from ram and base64 encoding to add to the kscrash report isn't really a good idea.

Reply:
I agree that it should be base64 encoded. We could only support attachments with a path, but I guess this would be confusing. The downside of using a temp file is that it causes lots of IO with our current approach. Each change of the global scope would cause IO because we would need to write the attachment to a temp file or check if it didn't change.

Reply:
@philipphofmann I'd build in another way that would require doing that. Like peeing a flag to whether it it's `dirty` or not.

Reply:
This approach would make sense for the whole Scope to SentryCrash synchronization. Hmm, maybe it makes sense to do this `dirty` flag first?

Reply:
Lets also discuss this tomorrow on our call?